examples/prorata/core/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+// Use below if we want to make the guest code no_std.
+// #![cfg_attr(not(test), no_std)]
+
+use csv;
+use rust_decimal::{Decimal, RoundingStrategy};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+
+#[derive(Serialize, Deserialize)]
+pub struct Recipient {
+    pub name: String,
+    #[serde(with = "rust_decimal::serde::str")]
+    pub share: Decimal,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Allocation {
+    pub name: String,
+    #[serde(with = "rust_decimal::serde::str")]
+    pub amount: Decimal,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct QueryResult {
+    pub allocation: Option<Allocation>,
+    pub total: Decimal,
+    pub csv_hash: Vec<u8>,
+}
+
+pub fn allocate(amount: Decimal, recipients: Vec<Recipient>) -> Vec<Allocation> {
+    /* Completely allocate the total funds `amount` by shares into dollars and
+    cents amounts for each recipient.
+
+    amount - total amount to distribute
+    recipients - list of recipients with their share of the total amount
+    */
```

Comment:
Prefer using `//` rather than multi-line comments

Reply:
Also this seems like a comment that would go just before the function, and perhaps even use `///`

Reply:
I'll read up on how to do this so it fits with the rust doc tooling.